source/common/grpc/google_async_client_impl.h

```diff
@@ -156,15 +245,24 @@ class GoogleAsyncStreamImpl : public AsyncStream,
     const bool end_stream_;
   };
 
+  GoogleAsyncTag init_tag_{*this, GoogleAsyncTag::Operation::Init};
+  GoogleAsyncTag read_initial_metadata_tag_{*this, GoogleAsyncTag::Operation::ReadInitialMetadata};
+  GoogleAsyncTag read_tag_{*this, GoogleAsyncTag::Operation::Read};
+  GoogleAsyncTag write_tag_{*this, GoogleAsyncTag::Operation::Write};
+  GoogleAsyncTag write_last_tag_{*this, GoogleAsyncTag::Operation::WriteLast};
+  GoogleAsyncTag finish_tag_{*this, GoogleAsyncTag::Operation::Finish};
+
   GoogleAsyncClientImpl& parent_;
+  GoogleAsyncClientThreadLocal& tls_;
+  // Latch our own version of this reference, so that completionThread() doesn't
+  // try and access via parent_, which might not exist in teardown.
```

Comment:
During tear down is the dispatcher itself guaranteed to still be valid? What protects against this case?

Reply:
Added comment; dispatchers outlive threads in the server loop.